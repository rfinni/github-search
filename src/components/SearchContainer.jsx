import React from 'react';
import PropTypes from 'prop-types';
import isEmpty from 'lodash/isEmpty';
import SearchField from './SearchField';
import Suggestions from './Suggestions';
import UserInfo from './UserInfo';

export const SearchContainer = ({
  isLoading,
  showSuggestions,
  suggestions,
  userData,
  inputValue,
  handleClearInput,
  handleChange,
  onClick,
}) => (
  <React.Fragment>
    <section className="search-wrapper">
      <SearchField
        isLoading={isLoading}
        value={inputValue}
        handleChange={handleChange}
        handleClear={handleClearInput}
      />

      {showSuggestions && (
        <Suggestions
          items={suggestions}
          handleClick={onClick}
        />
      )}
    </section>
    { !isEmpty(userData) && <UserInfo user={userData} /> }
  </React.Fragment>
);

SearchContainer.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  showSuggestions: PropTypes.bool.isRequired,
  suggestions: PropTypes.array.isRequired,
  userData: PropTypes.object.isRequired,
  inputValue: PropTypes.string.isRequired,
  handleClearInput: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default SearchContainer;
