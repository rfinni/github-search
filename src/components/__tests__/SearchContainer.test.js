import React from 'react';
import renderer from 'react-test-renderer';
import SearchContainer from '../SearchContainer';

describe('SearchContainer.jsx', () => {
  const props = {
    isLoading: false,
    showSuggestions: false,
    suggestions: [],
    userData: {},
    inputValue: '',
    handleClearInput: jest.fn(),
    handleChange: jest.fn(),
    onClick: jest.fn(),
  }

  it('should render snapshot correctly', () => {
    const wrapper = renderer.create(<SearchContainer {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
