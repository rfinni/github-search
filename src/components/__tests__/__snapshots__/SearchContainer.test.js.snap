// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer.jsx should render snapshot correctly 1`] = `
<section
  className="search-wrapper"
>
  <div
    className="search-field"
  >
    <input
      className="search"
      id="search"
      name="search"
      onChange={[Function]}
      placeholder="Search for a GitHub username"
      type="text"
      value=""
    />
  </div>
</section>
`;
